giphy_cli.py

```diff
@@ -1,31 +1,21 @@
-import click
-import requests
-
-API_KEY = "tFRkU1RvdbGFdp76FKKBuHvMRPpCrmM7"  # Use your API key
-BASE_URL = "https://api.giphy.com/v1/gifs/trending"
-
-@click.command()
-@click.option('--limit', default=10, help='Number of trending gifs to fetch.')
-def main(limit):
-    # Construct the API URL with the API key and limit
-    url = f"{BASE_URL}?api_key={API_KEY}&limit={limit}"
-
-    try:
-        # Make a GET request to the Giphy API
-        response = requests.get(url)
-
-        # Check if the request was successful
-        if response.status_code == 200:
-            # Parse the JSON response
-            data = response.json()
-
-            print(data)
-        else:
-            print(f"Request failed with status code: {response.status_code}")
-
-    except requests.exceptions.RequestException as e:
-        print(f"An error occurred: {e}")
-
-if __name__ == "__main__":
-    main()
-
+# Class will handle user input
+from giphy_api import GiphyAPI
+
+class GiphyCLI:
+    def __init__(self, api_key):
+        self.giphy_api = GiphyAPI(api_key)
+
+    def trending(self, limit=10, offset=0, rating='g'):
+        data = self.giphy_api.trending(limit, offset, rating)
+        self.print_gifs(data)
+
+    def search(self, query, limit=10, offset=0, rating='g', lang='en'):
```

Comment:
Same as above. The CLI interface should be identical to the interface you'll be exposing in Click.